File: packages/typescript/src/tools/DragAndDropTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";
import { field, type FieldMetadata } from "./Field.ts";

export class DragAndDropTool extends BaseTool {
  static description =
    "Drag one element onto another and drop it. Don't combine with HoverTool.";
  static fields: FieldMetadata[] = [
    field({
      name: "fromId",
      type: "integer",
      description: "Identifier (ID) of element to drag",
      paramName: "from_id",
    }),
    field({
      name: "toId",
      type: "integer",
      description: "Identifier (ID) of element to drop onto",
      paramName: "to_id",
    }),
  ];

  fromId: number;
  toId: number;

  constructor(args: { from_id: number; to_id: number }) {
    super();
    this.fromId = args.from_id;
    this.toId = args.to_id;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.dragAndDrop(this.fromId, this.toId);
  }
}

File: packages/python/src/alumnium/tools/drag_and_drop_tool.py
from pydantic import Field

from alumnium.drivers.base_driver import BaseDriver

from .base_tool import BaseTool


class DragAndDropTool(BaseTool):
    """Drag one element onto another and drop it. Don't combine with HoverTool."""

    from_id: int = Field(description="Identifier (ID) of element to drag")
    to_id: int = Field(description="Identifier (ID) of element to drop onto")

    def invoke(self, driver: BaseDriver):
        driver.drag_and_drop(self.from_id, self.to_id)
